### ISSUE TITLE
Windows 11 - Selected Toolbar Icons look light (bad) in the dark theme

### ISSUE THREAD

**Issue by vogella** [eclipse-platform/eclipse.platform.swt#217]
Eclipse SDK
Version: 2022-06 (4.25)
Build id: I20220620-1050
OS: Windows 11, v.10.0, x86_64 / win32
Java vendor: Eclipse Adoptium
Java runtime version: 17.0.2+8
Java version: 17.0.2

The selected toolbar items looks light (bad)in the dark theme under Windows 11. Does the native support need enhancement for this?

![image](https://user-images.githubusercontent.com/139910/174740514-9c76137f-14d0-4633-b24a-acd270cacf93.png)

**Comment by SyntevoAlex** [comment]
Doesn't look bad to me. Either way, our product doesn't use `Toolbar`, so I won't be able to work on it, sorry.

**Comment by vogella** [comment]
> Doesn't look bad to me. 

I highlighted the light buttons.

![image](https://user-images.githubusercontent.com/139910/174769760-a695b8a8-9189-4557-8271-cd4adc921d5e.png)

**Comment by SyntevoAlex** [comment]
Yes, I figured that you're talking about these buttons - yet, they don't look bad to me. Maybe not quite perfect :)

**Comment by jose-vasco** [comment]
They might not seem too bad in those very small versions with only an icon (which will potentially be barely visible anyway), but it gets really bad on bigger icons / ToolItems with text. Plus, this bright background behavior also applies on hover, not just when in selected state.

**Comment by Andr2ea** [comment]
The different background colors when hovering over the buttons is inconsistent, based on the type of the button. 
Dropdown:
<img width="1910" height="123" alt="Image" src="https://github.com/user-attachments/assets/8fc6c921-6b52-4cbc-9b17-bc34d7700b78" />

Button:
<img width="1891" height="111" alt="Image" src="https://github.com/user-attachments/assets/de2c2431-1828-4ae5-9546-e1ed1f5900e3" />

Could be considered as a feature for distinguishing the different functionalities, but the difference in contrast between them is pretty strong.

**Comment by Andr2ea** [comment]
I also see that the highlights/font in tree views for dark mode is a work in progress 
[https://github.com/eclipse-platform/eclipse.platform.ui/pull/3659](https://github.com/eclipse-platform/eclipse.platform.ui/pull/3659)

**Comment by xenogew** [comment]
My case the heap status bar also doesn't look good in dark mode.![IMG_20260310_112816.jpg](https://github.com/user-attachments/assets/870bde7d-3c5d-4893-be4d-1936d7d177f7)

**Comment by vogella** [comment]
Light theme (looks the same), left is old, right is new

<img width="2480" height="665" alt="Image" src="https://github.com/user-attachments/assets/52e7000e-9095-448f-a955-e205b95f1f8a" />

**Comment by vogella** [comment]
Dark theme, left is before the fix and right after the fix

<img width="2257" height="671" alt="Image" src="https://github.com/user-attachments/assets/7436e772-7bc5-47fd-afbf-495d02890ec0" />
